charts/core/src/Chart/hooks/updateUtils.ts

```diff
@@ -32,9 +30,38 @@ export function removeSeries(
   return updatedOptions;
 }
 
+/**
+ * Method to recursively merge two objects. It should update keys if they
+ * already exist and add them if they don't. However, it shouldn't completely
+ * overwrite a key it's an already existing object.
+ *
+ * They goal is to allow for partial updates to the chart options object.
+ */
+function recursiveMerge(
+  target: { [key: string]: any },
+  source: { [key: string]: any },
+) {
+  const updatedObj = { ...target };
+
+  for (const key in source) {
+    if (
+      typeof source[key] === 'object' &&
+      typeof updatedObj[key] === 'object'
+    ) {
+      // Recursively update nested objects
+      updatedObj[key] = recursiveMerge(updatedObj[key], source[key]);
+    } else {
+      // Update or add the value for the key
+      updatedObj[key] = source[key];
+    }
+  }
+
+  return updatedObj;
+}
+
 export function updateOptions(
   currentOptions: ChartOptions,
   options: Partial<ChartOptions>,
 ): Partial<ChartOptions> {
-  return merge({ ...currentOptions, ...options });
+  return recursiveMerge(currentOptions, options);
```

Comment:
Curious why we're replacing the lodash merge, but that's a question not a blocker

Reply:
The goal is to merge partial options objects in without overwriting previously set options that are unrelated to the update. This wasn't actually how `lodash.merge` worked (which I thought it would), leading to bugs when updating the same base keys with different options. I.e. when adding `splitLine` configurations on `xAxis` in order to add grid lines, then adding more axis specific config in the `XAxis` component related to the actual axis, the `splitLine` would be totally overwritten.

The updated test is a good example of what should pass and wasn't with `lodash.merge`:
```ts
  /**
   * Tests that option updates don't overwrite the entire chart options object.
   */
  test('should merge chart options non-destructively', () => {
    const currentOptions: Partial<ChartOptions> = {
      xAxis: {
        show: true,
        splitLine: {
          show: true,
        },
      },
    };
    const updatedOptions = updateOptions(currentOptions, {
      xAxis: {
        show: false, // This should only update the show property and not other properties
      },
      grid: {
        show: true,
      },
    });
    // @ts-ignore: Property 'show' does not exist on type 'Arrayable<AriaOption>'.
    expect(updatedOptions?.xAxis?.show).toBe(false);
    // @ts-ignore: Property 'show' does not exist on type 'Arrayable<AriaOption>'.
    expect(updatedOptions?.xAxis?.splitLine?.show).toBe(true);
    // @ts-ignore: Property 'show' does not exist on type 'Arrayable<GridOption>'.
    expect(updatedOptions?.grid?.show).toBe(true);
  });
```